Source/BugsnagClient.m

```diff
@@ -398,9 +394,16 @@ - (id)initWithConfiguration:(BugsnagConfiguration *)initConfiguration {
 
         // Start with a copy of the configuration metadata
         self.metadata = [[configuration metadata] deepCopy];
+        // sync initial state
         [self metadataChanged:self.configuration.metadata];
         [self metadataChanged:self.configuration.config];
         [self metadataChanged:self.state];
+
+        // add observers for future metadata changes
+        [self registerStateObserverWithBlock:^(BugsnagStateEvent *event) {
+            [self metadataChanged:event.data];
```

Comment:
`metadataChanged` has a debug message "Unknown metadata dictionary changed" if it's not one of the three it cares about. This could imply to future readers that it's being triggered incorrectly. Suggest we remove? Also we could check the event type is `kStateEventMetadata ` to further filter the calls?

Reply:
Removed log call to avoid confusion.